### PR TITLE
remove the ModelAverage、LookaheadOptimizer、FtrlOptimizer、DecayedAdagradOptimizer、DpsgdOptimizer and relocate the ExponentialMovingAverage、PipelineOptimizer、GradientMergeOptimizer

### PR DESCRIPTION
**Pull request by longranger2**
### PR types
Others

### PR changes
APIs

### Description
- [x] remove the ModelAverage in paddle/fluid/optimizer.py and use python/paddle/incubate/optimizer/modelaverage.py to replace it.
- [x] remove the LookaheadOptimizer in paddle/fluid/optimizer.py and use the LookAhead in paddle/incubate/optimizer/lookahead.py to replace it.
- [x] remove the FtrlOptimizer in paddle/fluid/optimizer.py.
- [x] remove the DecayedAdagradOptimizer in paddle/fluid/optimizer.py.
- [x] remove the DpsgdOptimizer in paddle/fluid/optimizer.py.
- [x] relocate the ExponentialMovingAverage in paddle/fluid/optimizer.py 
- [x] relocate the PipelineOptimizer in paddle/fluid/optimizer.py 
- [x] relocate the GradientMergeOptimizer in paddle/fluid/optimizer.py

### CODE REVIEW

**Comment by longranger2** [comment]
@zoooo0820 辛苦review下～

File: python/paddle/static/__init__.py [hunk 1]
```diff
@@ -95,7 +94,6 @@
     'name_scope',
     'program_guard',
     'WeightNormParamAttr',
-    'ExponentialMovingAverage',
```
Comment (zoooo0820) [inline python/paddle/static/__init__.py]: `paddle.static`属于公开API，这里保持现状吧

File: test/legacy_test/test_ema_fleet.py [hunk 2]
```diff
@@ -56,7 +56,9 @@ def setUp(self):
                 )
                 optimizer.minimize(cost)
 
-                self._ema = static.ExponentialMovingAverage(self._ema_decay)
```
Comment (zoooo0820) [inline test/legacy_test/test_ema_fleet.py]: 这里仍用`static.ExponentialMovingAverage`的调用方式，不建议使用新的路径

**Comment by longranger2** [comment]
@zoooo0820 已经修复了，辛苦review下～

**Comment by longranger2** [comment]
@zoooo0820 辛苦review下～

**Review by zoooo0820 — APPROVED** [review]
LGTM

**Review by XieYunshen — DISMISSED** [review]
LGTM 
单测删除

**Comment by longranger2** [comment]
已经修改好了，辛苦review下～

**Comment by longranger2** [comment]
@zoooo0820 @XieYunshen 已经修改好了，辛苦review下～

**Review by XieYunshen — DISMISSED** [review]

**Review by XieYunshen — APPROVED** [review]

**Review by XieYunshen — DISMISSED** [review]

File: python/paddle/static/nn/__init__.py [hunk 3]
```diff
@@ -40,6 +40,7 @@
 
 from .common import embedding  # noqa: F401
 from .common import sparse_embedding  # noqa: F401
+from .common import ExponentialMovingAverage  # noqa: F401
```
Comment (jeff41404) [inline python/paddle/static/nn/__init__.py]: need `import ExponentialMovingAverage` here?

Comment (longranger2) [reply python/paddle/static/nn/__init__.py]: Actually, dont't need

**Review by XieYunshen — APPROVED** [review]

**Pull request closed without merging**